### PR TITLE
[XCBuildConfiguration] Fix infinite recursion in #resolve_build_setting

### PR DESCRIPTION
**Pull request by Ruenzuo**
This should fix the infinite recursion case reported in https://github.com/CocoaPods/Xcodeproj/issues/511

### CODE REVIEW

**Comment by dnkoutso** [comment]
Need a changelog entry

**Comment by CocoaPodsBarista** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Please include a CHANGELOG entry to credit yourself! <br />
You can find it at <a href="https://github.com/CocoaPods/CocoaPods/blob/master/CHANGELOG.md">CHANGELOG.md</a>.</td>
    </tr>
  </tbody>
</table>

Here's an example of your CHANGELOG entry:

```markdown
* [XCBuildConfiguration] Fix infinite recursion in #resolve_build_setting  
  [Ruenzuo](https://github.com/Ruenzuo)
  [#issue_number](https://github.com/CocoaPods/CocoaPods/issues/issue_number)
```

*note*: There are two invisible spaces after the entry's text.


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by Ruenzuo** [comment]
@dnkoutso thank you for super fast the review ☺️

File: CHANGELOG.md [hunk 1]
```diff
@@ -11,6 +11,9 @@
 * Verify container portal when checking dependency target membership.  
   [izaakschroeder](https://github.com/izaakschroeder)
   [#513](https://github.com/CocoaPods/Xcodeproj/issues/513)
+* [XCBuildConfiguration] Fix infinite recursion in #resolve_build_setting
```
Comment (dnkoutso) [inline CHANGELOG.md]: nit extra line here

Comment (Ruenzuo) [reply CHANGELOG.md]: It has two spaces after the last character. Does it need anything else?

Comment (dnkoutso) [reply CHANGELOG.md]: I meant from above, separate the two changelog entries

Comment (dnkoutso) [reply CHANGELOG.md]: also as a nit wrap the `#resolve_build_setting` with backticks

Comment (Ruenzuo) [reply CHANGELOG.md]: Alright, got it!

**Comment by tomaskraina** [comment]
Any estimate on when this is going to be released?

**Comment by orta** [comment]
With the next CocoaPods beta 👍

**Pull request merged**